### ISSUE TITLE
Add data binding for markdown widget

### ISSUE THREAD

**Issue by Bluebugs** [fyne-io/fyne#2835]
### Is your feature request related to a problem? Please describe:
It would be nice to have data binding for rich text via a markdown binding.

### Is it possible to construct a solution with the existing API?
Not currently doable.

### Describe the solution you'd like to see:
I can see the following:
```
func NewRichTextFromMarkdownWithData(data binding.String) *RichText
func (t *RichText) BindMarkdown(data binding.String)
func (t *RichText) UnbindMarkdown()
```

**Comment by andydotxyz** [comment]
Hmm, not sure about this one. The widget is `RichText` not `Markdown`, and it's a convenience import - the "FromXxx" indicates that a conversion takes place.

**Comment by Bluebugs** [comment]
Yes, finding the right name is a bit tricky here. I definitively see the use case for this, but don't know about the name. What about: `NewRichTextWithDataMarkdown` .

**Comment by andydotxyz** [comment]
I don't think this is about naming... the `RichTextFromMarkdown` is not something that is working with markdown, it is a markdown parser that creates a `RichText` which is the main widget.
It's a bit like binding an `ImageFromURI` and expecting the setter to write out the image data to source server when it is modified in the app.

From a technological point we have no code that serialises our rich text widget to Markdown, as it supports things that Markdown does not...

**Comment by Bluebugs** [comment]
I see your point. I guess we would need something like a MarkdownEntry so that it can be a two way binding for it. I was thinking of using just one way, but indeed two way make more sense as it seems doable. I guess this is depending on #21 then.

**Comment by andydotxyz** [comment]
I don't think I follow. Whether the user can edit the value or not is not what I meant (RichText can be mutated in code).
The problem is that Markdown is parsed to create a RichTextSegment tree that controls how the widget looks - we have no code that then somehow serialises the same tree back into Markdown.

**Comment by andydotxyz** [comment]
I was wondering if this issue is to stay open? Like I described above we can't bind to something that is an intermediate format where we only support read.

Is this becoming a feature request to write a (lossy) markdown document from a `RichText` widget? I would recommend against binding to data that goes through such a process. 

Equally there have been many demonstrations of how you can "bind" the markdown rendering to an entry with simple `OnChanged` usage which seems to satisfy the deliverable portion of the connection described.

**Comment by Bluebugs** [comment]
I think we can close this for the moment and revisit later.
